app/components/documents/document-card.js

```diff
@@ -205,6 +207,21 @@ export default class DocumentsDocumentCardComponent extends Component {
     this.loadAccessLevelRelatedData.perform();
     this.loadPublicationFlowRelatedData.perform();
     this.loadSignatureRelatedData.perform();
+    this.loadAccessPillVisibilityData.perform();
+  }
+
+  @task 
+  *loadAccessPillVisibilityData() {
+    const decisionActivity = 
+    this.args.decisionActivity ? this.args.decisionActivity : 
+      yield this.store.queryOne('decision-activity', {
+        'filter[treatment][agendaitems][pieces][:id:]': this.args.piece.id
+      })
```

Comment:
I wouldn't try to query `decisionActivity` here.
Case and point > a subcase with 20 docs > 20 queries (all different) vs 1 passing of an already loaded `decisionActivity` to `<Documents::DocumentCardList>`

We already have logic in most places to show the alerts, could we do something there?
We could just make a new params for document-card (`showAccessLevel`) and set it to false

Reply:
right now, we don't pass `decisionActivity` for subcase documents, but there can be more than 1.
We should only check the resultCode of the latest one? or check if there is atleast 1 approved/acknowledged one.

afaik `agendaItemTreatment` and `decisionActivity` is not propagated until after release of decisions for other profiles but some documents already are, how are we going to check decision status in that case? Need to look into that.
If we have docs but no `decisionActivity` > also hide the accessLevels? 
I think we haven't thought of this tbh.
But it will be impossible to check decision status for other profiles prior to releasing decisions.

Reply:
so profiles will see all documents that are allowed + the access levels right. When we finally release decision, they will be able to see less if the item is postponed, which is weird.

Reply:
This thing is about "uitgestelde agendapunten", specifically ones that are re-used on a later agenda (because users on the original agenda will see an postponed agendaitem with documents that are seemingly readable.

While yes, there's a window where an agendaitem can get postponed and before the agenda is finalized ministers are allowed to see the agendaitem and the documents, it's very unlikely that it will have already been put on a later agenda (and thus, the access level being changed). So I don't think it really matters here.

Reply:
As it's implemented now, the accessLevelPill does idd not show on agenda's where the decisions haven't been released yet, which is not what was intended here.

Sergio's point is valid, so I think it's more important to show the pill if we can't be sure than to hide it